crates/gfx-plugin/src/rendering.rs

```diff
@@ -0,0 +1,224 @@
+}
+
+/// An error in construction of a rendered entity.
+#[derive(Error, Debug)]
+pub enum RenderedEntityBuilderError {
+    /// Failed to create a new entity.
+    #[error("failed to create a new entity")]
+    EntityCreation(#[source] Box<dyn Error + Send + Sync>),
+    /// Failed to add component to entity.
+    #[error("failed to add component to entity")]
+    ComponentAdding(#[source] Box<dyn Error + Send + Sync>),
+}
+
+/// Whether an operation on the graphical application succeeded.
+pub type RenderedEntityBuilderResult<T, E = RenderedEntityBuilderError> = Result<T, E>;
+
+/// Builds a visible (i.e. rendered) entity.
+#[derive(Debug)]
+pub struct RenderedEntityBuilder<'app, App> {
+    application: &'app mut App,
+    entity: Entity,
+    model: Model,
+    position: Option<Position>,
+    rotation: Option<Rotation>,
+    scale: Option<Scale>,
+}
+
+impl<InnerApp: Application> GraphicalApplication<InnerApp> {
+    /// Constructs a new [`RenderedEntityBuilder`] which can be used to build
+    pub fn rendered_entity_builder(
+        &mut self,
+        model: Model,
+    ) -> RenderedEntityBuilderResult<RenderedEntityBuilder<InnerApp>> {
+        RenderedEntityBuilder::new(&mut self.application, model)
+    }
+}
+
+impl<'app, App: Application> RenderedEntityBuilder<'app, App> {
+    /// Creates a new instance of the builder.
+    ///
+    /// Note: the entity is created immediately in this call.
+    pub fn new(application: &'app mut App, model: Model) -> RenderedEntityBuilderResult<Self> {
+        let entity = application
+            .create_entity()
+            .map_err(|error| RenderedEntityBuilderError::EntityCreation(Box::new(error)))?;
+
+        Ok(RenderedEntityBuilder {
+            application,
+            entity,
+            model,
+            position: None,
+            rotation: None,
+            scale: None,
+        })
+    }
+
+    /// Sets the position of the entity.
+    pub fn with_position(mut self, position: Position) -> Self {
+        self.position = Some(position);
+        self
+    }
+
+    /// Sets the rotation of the entity.
+    pub fn with_rotation(mut self, rotation: Rotation) -> Self {
+        self.rotation = Some(rotation);
+        self
+    }
+
+    /// Sets the scale of the entity.
+    pub fn with_scale(mut self, scale: Scale) -> Self {
+        self.scale = Some(scale);
+        self
+    }
+
+    /// Finishes the building of the rendered entity and returns it, if successful.
+    pub fn build(self) -> RenderedEntityBuilderResult<Entity> {
+        let RenderedEntityBuilder {
+            application,
+            entity,
+            model,
+            position,
+            rotation,
+            scale,
+        } = self;
+
+        application
+            .add_component(entity, model)
+            .map_err(to_component_adding_error)?;
+
+        add_component_or_default(application, position, entity)?;
```

Comment:
I've been thinking about whether it would be useful for a `Read`/`Write` system parameter that matches any entity and returns an `Option` with the component. In this case it would be useful so you don't have to add a default component. Do you think adding this system parameter would be a good addition?

Reply:
Ohh, so we could have the system query be `rendering_system(..., maybe_rotation: Read<Option<Rotation>>)` and then use a default value if it's `None`? That'd be really nice. 

If it's not a lot of trouble to add then I think it would be a great addition, though I am a bit concerned about how it would affect cache locality. Wouldn't there need to be jumping between several different archetypes to do this? But maybe it's few enough jumps that it's negligible.

Reply:
> Implementing this should be easy, but dynamic archetypes must be implemented first.

Great, would you mind creating an issue for that with the requirements and dependencies needed, then?